tests/test_project_caching.py

```diff
@@ -408,6 +408,23 @@ def setup_workspace_and_west_update(workspace, foo_head, bar_head):
     for msg in msgs:
         assert msg in stdout
 
+    # Rename remote default branches and update remote (results in dangling remote HEAD).
+    with chdir(foo_remote):
+        subprocess.check_call([GIT, 'branch', '-m', 'master', 'renamed-master'])
+    with chdir(bar_remote):
+        subprocess.check_call([GIT, 'branch', '-m', 'master', 'changed/main'])
+    with chdir(auto_cache_dir_foo):
+        subprocess.check_call([GIT, 'remote', 'update', '--prune'])
+    with chdir(auto_cache_dir_bar):
+        subprocess.check_call([GIT, 'remote', 'update', '--prune'])
```

Comment:
No need to use `chdir`

```suggestion
    subprocess.check_call([GIT, 'branch', '-m', 'master', 'renamed-master'], cwd=foo_remote)
    subprocess.check_call([GIT, 'branch', '-m', 'master', 'changed/main'], cwd=bar_remote)
    subprocess.check_call([GIT, 'remote', 'update', '--prune'], cwd=auto_cache_dir_foo)
    subprocess.check_call([GIT, 'remote', 'update', '--prune'], cwd=auto_cache_dir_bar)
```